lib/verifier/post_zephir_verifier.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: true
+
+require "zlib"
+require_relative "../verifier"
+require_relative "../derivatives"
+
+# Verifies that post_zephir workflow stage did what it was supposed to.
+
+# TODO: document and verify the files written by monthly process.
+# They should be mostly the same but need to be accounted for.
+
+module PostZephirProcessing
+  class PostZephirVerifier < Verifier
+    attr_reader :current_date
+
+    def run_for_date(date:)
+      @current_date = date
+      verify_catalog_archive
+      verify_catalog_prep
+      verify_dollar_dup
+      verify_ingest_bibrecords
+      verify_rights
+      verify_zephir_data
+    end
+
+    # Frequency: ALL
+    # Files: CATALOG_ARCHIVE/zephir_upd_YYYYMMDD.json.gz
+    #   and potentially CATALOG_ARCHIVE/zephir_full_YYYYMMDD_vufind.json.gz
+    # Contents: ndj file with one catalog record per line
+    # Verify:
+    #   readable
+    #   line count must be the same as input JSON
+    def verify_catalog_archive(date: current_date)
+      zephir_update_derivative_params = {
+        location: :CATALOG_ARCHIVE,
+        name: "zephir_upd_YYYYMMDD.json.gz",
+        date: date
+      }
+      zephir_update_path = self.class.dated_derivative(**zephir_update_derivative_params)
+      verify_file(path: zephir_update_path)
+      verify_parseable_ndj(path: zephir_update_path)
+
+      if date.last_of_month?
+        zephir_full_derivative_params = {
+          location: :CATALOG_ARCHIVE,
+          name: "zephir_full_YYYYMMDD_vufind.json.gz",
+          date: date
+        }
+
+        ht_bib_export_derivative_params = {
+          location: :ZEPHIR_DATA,
+          name: "ht_bib_export_full_YYYY-MM-DD.json.gz",
+          date: date
+        }
+        output_path = self.class.dated_derivative(**zephir_full_derivative_params)
+        verify_file(path: output_path)
+        verify_parseable_ndj(path: output_path)
+        output_linecount = gzip_linecount(path: output_path)
+
+        input_path = self.class.dated_derivative(**ht_bib_export_derivative_params)
+        verify_file(path: input_path)
+        verify_parseable_ndj(path: input_path)
+        input_linecount = gzip_linecount(path: input_path)
+
+        if output_linecount != input_linecount
+          error(
+            message: sprintf(
+              "output line count (%s = %s) != input line count (%s = %s)",
+              output_path,
+              output_linecount,
+              input_path,
+              input_linecount
+            )
+          )
+        end
+      end
+    end
+
+    # Frequency: ALL
+    # Files: CATALOG_PREP/zephir_upd_YYYYMMDD.json.gz and CATALOG_PREP/zephir_upd_YYYYMMDD_delete.txt.gz
+    #   and potentially CATALOG_PREP/zephir_full_YYYYMMDD_vufind.json.gz
+    # Contents:
+    #   json.gz files: ndj with one catalog record per line
+    #   delete.txt.gz: see `#verify_deletes_contents`
+    # Verify:
+    #   readable
+    #   TODO: deletes file is combination of two component files in TMPDIR?
```

Comment:
Need to address TODO

Reply:
don't understand TODO

Reply:
@moseshll Anything you were expecting to do here or should we remove that TODO?

Reply:
To explain the TODO: @moseshll reports that part of the deletes file comes from Zephir and part from postZephir.pm. The two deletes get glommed together into the final product. There may be some research to verify this is happening correctly and see if that's even possible to do.

Reply:
To address later: #57